examples/gno.land/r/sys/validators/v2/poc.gno

```diff
@@ -16,7 +28,22 @@ func NewPropRequest(changesFn func() []validators.Validator, title, description
 	}
 
 	callback := func(cur realm) error {
-		for _, change := range changesFn() {
+		// Enforce cooldown: proposals can only update the validator set after the cooldown period.
+		if time.Since(lastValsetUpdate) < valsetUpdateCooldown {
+			return errors.New(errValsetUpdateCooldown)
+		}
+
+		// Get the proposed validator changes.
+		proposedChanges := changesFn()
+
+		// Calculate the maximum allowed changes (less than 2/3 of current validator set).
+		maxAllowedChanges := ((vp.LenValidators() * 2) / 3)
+		// If there are more than 1 validator, limit the number of changes.
+		if maxAllowedChanges > 0 && len(proposedChanges) >= maxAllowedChanges {
```

Comment:
shouldn't this be weighted by voting power?

Reply:
not really sure. I think it number of validator not the  sigma(weight). cc @Nemanya8  to confirm this

Reply:
humm it seems you have right https://ibc.cosmos.network/main/ibc/light-clients/tendermint/overview/